### PR TITLE
New <Counter /> example

### PR DESCRIPTION
**Pull request by jmpas**
## Todo
- [x] Component
- [x] Stories
- [x] Specs
## Changes
- `babel-cli` as local dev dependency
- `Foo` example removed
- `Counter` example added

### CODE REVIEW

File: src/Counter.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        this.setState({ count: this.state.count - 1 });
+        this.isDone();
+    }
+
+    render() {
+        return <span>{this.state.count}</span>;
+    }
+}
+
+Counter.propTypes = propTypes;
```
Comment (willianjusten) [inline src/Counter.js:53]: Why not to put the object directly? We don't need to create one more object each time.

```
Counter.propTypes = {
  start: React.PropTypes.number.isRequired,
  end: React.PropTypes.number.isRequired,
  done: React.PropTypes.func,
}
```

Nevermind...this is the correct way following Airbnb Styleguide.

Comment (jmpas) [inline src/Counter.js:53]: @willianjusten, I thought about that as well, but I decided to follow [this](https://github.com/airbnb/javascript/tree/master/react#ordering)

File: src/Counter.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
```
Comment (willianjusten) [inline src/Counter.js]: Import { PropTypes }  here.

File: src/Counter.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+const propTypes = {
+    start: React.PropTypes.number.isRequired,
```
Comment (willianjusten) [inline src/Counter.js]: Remove "React", use the PropTypes directly instead.

**Comment by jmpas** [comment]
@willianjusten hey lad, have a look, would ya?

screen shot 2016-07-21 at 10 57 55 pm

**Comment by willianjusten** [comment]
Hey fella, don't try to impress me with coverage stats. I'll take a look.

File: package.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
   "description": "An opinionated structure for reusable and declouped react components.",
   "main": "dist/Foo.js",
   "scripts": {
-    "build": "npm run clean && babel src -d dist",
+    "build": "npm run clean && ./node_modules/.bin/babel src -d dist",
```
Comment (willianjusten) [inline package.json:7]: ops xD

File: tests/specs/Counter.spec.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import chai, { expect } from 'chai';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import { shallow, mount } from 'enzyme';
+import Counter from '../../src/Counter';
```
Comment (willianjusten) [inline tests/specs/Counter.spec.js]: Put this at the bottom...

**Comment by willianjusten** [comment]
Nice =D

**Pull request merged**